darts/tests/dataprocessing/transformers/test_window_transformations.py

```diff
@@ -0,0 +1,416 @@
+                "window": [5],
+            }  # window list
+            self.series_univ_det.window_transform(transforms=window_transformations)
+
+        with self.assertRaises(ValueError):
+            window_transformations = {
+                "function": "mean",
+                "window": 3,
+                "mode": "rolling",
+                "step": -2,
+            }  # Negative step
+            self.series_univ_det.window_transform(transforms=window_transformations)
+
+        with self.assertRaises(ValueError):
+            window_transformations = {
+                "function": "mean",
+                "window": 3,
+                "mode": "rnd",
+            }  # invalid mode
+            self.series_univ_det.window_transform(transforms=window_transformations)
+
+        with self.assertRaises(ValueError):
+            window_transformations = {
+                "function": "mean",
+                "mode": "rolling",
+                "window": 3,
+                "center": "True",
+            }  # forecating_safe=True vs center=True
+            self.series_univ_det.window_transform(transforms=window_transformations)
+
+    def test_ts_windowtransf_output_series(self):
+        # univariate deterministic input
+        window_transformations = {"function": "sum", "mode": "rolling", "window": 1}
+        transformed_ts = self.series_univ_det.window_transform(
+            transforms=window_transformations
+        )
+
+        self.assertEqual(
+            list(itertools.chain(*transformed_ts.values().tolist())),
+            list(itertools.chain(*self.series_univ_det.values().tolist())),
+        )
+        self.assertEqual(len(transformed_ts.components), 1)
+
+        # multivariate deterministic input
+        # transform one component
+        window_transformations = {
+            "function": "sum",
+            "mode": "rolling",
+            "window": 1,
+            "components": "0",
+        }
+
+        transformed_ts = self.series_multi_det.window_transform(
+            transforms=window_transformations
+        )
+        self.assertEqual(len(transformed_ts.components), 1)
+
+        transformed_ts = self.series_multi_det.window_transform(
+            transforms=window_transformations, keep_non_transformed=True
+        )
+        self.assertEqual(len(transformed_ts.components), 5)
+
+        # transform multiple components
+        window_transformations = {
+            "function": "sum",
+            "mode": "rolling",
+            "window": 1,
+            "components": ["0", "0_1"],
+        }
+
+        transformed_ts = self.series_multi_det.window_transform(
+            transforms=window_transformations
+        )
+        self.assertEqual(len(transformed_ts.components), 2)
+
+        transformed_ts = self.series_multi_det.window_transform(
+            transforms=window_transformations, keep_non_transformed=True
+        )
+        self.assertEqual(len(transformed_ts.components), 6)
```

Comment:
(nitpicking) it would be a tiny bit better to test the actual component names (incl the original ones)

Reply:
not sure I understand, I proposed something based on what I understood. Using sets to work with the column names was messing up the order in fact, good thing you asked me to check the component names.